jwst/cube_build/data_types.py

```diff
@@ -89,7 +89,7 @@ def __init__(self, input, single, output_file, output_dir):
             self.output_name = 'Temp'
             self.input_models = input_models
             if not single:  # find the name of the output file from the association
-                self.output_name = input_models.meta.asn_table.products[0].name
+                self.output_name = input_models.asn_table["products"][0]["name"]
```

Comment:
This line will crash with a KeyError if CubeBuildStep is called on a ModelContainer made from a list of models or filenames, instead of from an association.  It looks like there's a similar crash for the implementation on main as well, but it might be useful instead to add a function to ModelContainer to robustly get ASN metadata or throw a better error if not possible.

Reply:
I tried fixing this by making the default asn_table contain the correct keys, but all with None values.  I haven't tested this yet but will do so.

Reply:
This gets further than the previous version, but still crashes.

I'm testing using regression test data from nirspec/ifu:
```
import glob
from jwst import datamodels
from jwst.cube_build import CubeBuildStep

fn = glob.glob('jw01249*_cal.fits')
container = datamodels.ModelContainer(fn)
CubeBuildStep.call(container)   #  or CubeBuildStep.call(fn) - both crash the same way
```

It now crashes when trying to make an output name:
```
--> 216     suffix = self.output_name_base[self.output_name_base.rfind('_') + 1:]
    217     if suffix in ['g140m', 'g235m', 'g395m', 'g140h', 'g235h', 'g395h', 'prism']:
    218         self.output_name_base = self.output_name_base[:self.output_name_base.rfind('_')]

AttributeError: 'NoneType' object has no attribute 'rfind'
```
If it's not easy to make a real default output name here, we can just catch the None when the output_filename is set and throw an error saying that ASN metadata is required.

Reply:
I tested this in the same way on the most recent commit and it runs successfully, feel free to try it yourself though

Reply:
That's working now, thanks!